### PR TITLE
ncspot: update to 1.2.1

### PR DESCRIPTION
**Pull request by terminaldweller**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc

### CODE REVIEW

**Comment by xavierfortier1** [comment]
I came across this PR while I was looking in the repo and just so you know, I'm trying to update `spotify-player` in #53498 and there is a problem in the `aws-lc-rs` crate. You also have to update the `rust-binden` template to match the latest version, cause .

Also, lint checks doesn't pass, because you have to amend and force push the changes instead of creating a new commit (so, only a commit per update/package).

**Comment by xavierfortier1** [comment]
Hello! There has been updates in #53498. I've managed to build the package with a workaround for `armv7l` provided by `aws`. Take a look at the template, you can probably testusethe new dependencies and the workaround for `ncspot`!

File: srcpkgs/ncspot/template [hunk 1]
```diff
@@ -32,3 +32,8 @@ post_install() {
 	vcompletion misc/_ncspot zsh
 	vcompletion misc/ncspot.fish fish
 }
+
+# FIXME: Workaround. Should be removed when https://github.com/aws/aws-lc-rs/issues/632 is resolved.
+if [ "$XBPS_TARGET_MACHINE" = armv7l ]; then
+	export AWS_LC_SYS_CFLAGS="-Wno-stringop-overflow"
+fi
```
Comment (classabbyamp) [inline srcpkgs/ncspot/template:39]: this if block should go before functions

Comment (xavierfortier1) [reply srcpkgs/ncspot/template]: This if is not enough, because armv6l won't compile. This is why I proposed the case like the template of spotify-player

Comment (classabbyamp) [reply srcpkgs/ncspot/template]: that too

File: srcpkgs/ncspot/template [hunk 2]
```diff
@@ -14,7 +14,14 @@ license="BSD-2-Clause"
...
 distfiles="https://github.com/hrkfdn/ncspot/archive/refs/tags/v${version}.tar.gz"
-checksum=d3cd828cebb3e84470f03be16925db566d4dd8289cfd3e230e64278ec9d96338
+checksum=6bd08609a56aa5854a1964c9a872fe58b69a768d7d94c874d40d7a8848241213
+
+# FIXME: Workaround will be removed when https://github.com/aws/aws-lc-rs/issues/632 is resolved
+	case "$XBPS_TARGET_MACHINE" in
+	armv[67]l)
+	export AWS_LC_SYS_CFLAGS="-Wno-stringop-overflow"
+;;
+esac
```
Comment (classabbyamp) [inline srcpkgs/ncspot/template:24]:
```suggestion
case "$XBPS_TARGET_MACHINE" in
	armv[67]l)
		export AWS_LC_SYS_CFLAGS="-Wno-stringop-overflow"
		;;
esac
```

Comment (classabbyamp) [reply srcpkgs/ncspot/template]: still an extra indent on the first line of the case statement

**Pull request merged**